### PR TITLE
enh(context): Integrate Jotai to Resources Status Details

### PR DESCRIPTION
**Pull request by Thebarda**
## Description

This almost replaces the React Context by Jotai. In this PR. This is the first PR in which I replace the details related contexts data in RS.

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [x] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 20.04.x
- [ ] 20.10.x
- [ ] 21.04.x
- [ ] 21.10.x
- [x] 22.04.x

<h2> How this pull request can be tested ? </h2>

Everything in the details panel should work as before

## Checklist

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

**Review by Jordan91 — DISMISSED** [review]

File: www/front_src/src/Resources/Graph/Performance/Graph/annotationsAtoms.ts [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+  mouseX: number;
+  resourceId: string;
+  timeline: Array<TimelineEvent> | undefined;
+  xScale: ScaleTime<number, number>;
+}
+
+interface GetIsBetweenProps {
+  xEnd: number;
+  xStart: number;
+}
```
Comment (bdauria) [inline www/front_src/src/Resources/Graph/Performance/Graph/annotationsAtoms.ts:38]:
```suggestion
interface TimeInterval {
  xEnd: number;
  xStart: number;
}
```

we can then get rid of the 'x' prefices

File: www/front_src/src/Resources/Graph/Performance/Graph/annotationsAtoms.ts [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+
+interface GetColorProps {
```
Comment (bdauria) [inline www/front_src/src/Resources/Graph/Performance/Graph/annotationsAtoms.ts:40]:
```suggestion
interface AnnotationColor {
```

File: www/front_src/src/Resources/Graph/Performance/Graph/annotationsAtoms.ts [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+  annotation: AnnotationAtom;
+  color: string;
+}
+
+interface AnnotationAtom {
+  event?: TimelineEvent;
+  resourceId: string;
+}
+
+interface GetIsNotHoveredOrNilProps {
```
Comment (bdauria) [inline www/front_src/src/Resources/Graph/Performance/Graph/annotationsAtoms.ts:50]:
```suggestion
interface AnnotationWithHoveredState {
```

File: www/front_src/src/Resources/Graph/Performance/Graph/mouseTimeValueAtoms.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
+import { atom } from 'jotai';
+import { isNil } from 'lodash';
+import { not } from 'ramda';
+
+import { TimeValue } from '../models';
+
+export type MousePosition = [number, number] | null;
+
+interface ChangeMousePositionAndTimeValueProps {
```
Comment (bdauria) [inline www/front_src/src/Resources/Graph/Performance/Graph/mouseTimeValueAtoms.ts:9]:
```suggestion
interface PositionTimeValue {
```

File: www/front_src/src/Resources/Graph/Performance/Graph/mouseTimeValueAtoms.ts [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+  position: MousePosition;
+  timeValue: TimeValue | null;
+}
+
+interface ChangeTimeValueProps {
```
Comment (bdauria) [inline www/front_src/src/Resources/Graph/Performance/Graph/mouseTimeValueAtoms.ts:14]:
```suggestion
interface NewTimeValueInViewportState {
```

**Review by bdauria — DISMISSED** [review]

**Review by jdelpierre — DISMISSED** [review]

**Review by bdauria — APPROVED** [review]

**Review by jdelpierre — APPROVED** [review]

**Pull request merged**